src/rulesets/evaluators.ts

```diff
@@ -1,12 +1,79 @@
-import { Optional } from '@stoplight/types';
+import { join, stripRoot } from '@stoplight/path';
+import { Dictionary, Optional } from '@stoplight/types';
 import { isObject } from 'lodash';
 import { IFunction, JSONSchema } from '../types';
 import { decorateIFunctionWithSchemaValidation } from './validation';
 
-export type CJSExport = Partial<{ exports: object | ESCJSCompatibleExport }>;
+export type CJSExport = Partial<{ exports: object | ESCJSCompatibleExport; require: NodeJS.Require }>;
 export type ESCJSCompatibleExport = Partial<{ default: unknown }>;
 export type ContextExport = Partial<{ returnExports: unknown }>;
 
+function requireUnavailable() {
+  throw new ReferenceError('require() is supported only in the Node.JS environment');
+}
+
+function stubRequire(): NodeJS.Require {
+  function req() {
+    requireUnavailable();
+  }
+
+  const descriptors: Dictionary<PropertyDescriptor, keyof NodeJS.Require> = {
+    resolve: {
+      enumerable: true,
+      get: requireUnavailable,
+    },
+
+    main: {
+      enumerable: true,
+      get: requireUnavailable,
+    },
+
+    cache: {
+      enumerable: true,
+      get: requireUnavailable,
+    },
+
+    extensions: {
+      enumerable: true,
+      get: requireUnavailable,
+    },
+  };
+
+  return Object.defineProperties(req, descriptors);
+}
+
+function proxyRequire(source: string): NodeJS.Require {
+  const actualRequire = require;
+  function req(p: string) {
+    if (p.startsWith('.')) {
+      p = join(source, '..', stripRoot(p));
+    }
+
+    return actualRequire.call(null, p);
+  }
+
+  return Object.defineProperties(req, Object.getOwnPropertyDescriptors(actualRequire));
+}
+
+const isRequiredSupported =
+  typeof require === 'function' &&
+  typeof require.main === 'object' &&
+  require.main !== null &&
+  'paths' in require.main &&
+  'cache' in require;
+
+const createRequire = (source: string | null): NodeJS.Require => {
+  if (!isRequiredSupported) {
+    return stubRequire();
+  }
+
+  if (source === null) {
+    return require;
+  }
+
+  return proxyRequire(source);
+};
+
```

Comment:
@P0lip @philsturgeon 

This parts suggest that Spectral is going to load whatever code, assuming it's all in good intent. Practically, I could write a rule that claims to lint a part of a JSON document but on the other hand it's formatting my disk or sending my credentials.

I know that in the browser is not an issue, but for anybody deploying a Spectral on the server this could be a problem. I highly suggest to use [vm2](https://github.com/patriksimek/vm2) and to put a clear whitelist of the modules that can be used.

Disabling external requires would be great, disabling `fs` is probably be a good start. The best tradeoff would be to propagate these configuration values from the spectral configuration file so that the user will decide how much risk they want to put in an external rule.

Reply:
//cc @nulltoken

Reply:
We've had a couple of discussions regarding sandboxing. I was also leaning towards running the code in an isolated environment, but eventually we decided not to.
The primary reason is that no linter really does it. It's kind of the same story as with dependencies. We all kind of trust them. So we do here.

Reply:
https://stoplight-internal.slack.com/archives/CEUN2664A/p1581573511007600?thread_ts=1581573511.007600&cid=CEUN2664A

Reply:
Read the conversation. Yeah my point is — as long you run in from your computer I really do not mind. My concern is more if Spectral will be run on a server (Stoplight SaaS?) then it might become an issue.

Since we're not doing this at the moment I guess it's ok to sandbox this. Thanks for checking it out!

Reply:
>  My concern is more if Spectral will be run on a server (Stoplight SaaS?) then it might become an issue.

The plan is to spawn Spectral instances in a worker, so we'll also benefit from that, as workers are already partly isolated.
In any case, we are not planning to run node in a privileged mode (as root on Unix OSses, admin rights on Windows), so this will also mitigate certain issues.
Either way, point taken. I believe it'll be really worthwhile to revisit it at some point.
Providing a sandbox, for instance as an extra feature, would be something parts of users would likely benefit from.